winsup/utils/kill.cc

```diff
@@ -162,7 +163,10 @@ forcekill (int pid, int sig, int wait)
   if (pid > 0)
     p = (external_pinfo *) cygwin_internal (CW_GETPINFO_FULL, pid);
   DWORD dwpid = p ? p->dwProcessId : (DWORD) pid;
-  HANDLE h = OpenProcess (PROCESS_TERMINATE, FALSE, (DWORD) dwpid);
+  HANDLE h = OpenProcess (PROCESS_CREATE_THREAD | PROCESS_QUERY_INFORMATION |
+			  PROCESS_VM_OPERATION | PROCESS_VM_WRITE |
+			  PROCESS_VM_READ | PROCESS_TERMINATE,
+			  FALSE, (DWORD) dwpid);
   if (!h)
```

Comment:
I agree with Jeff here. It might make more sense to move the `OpenProcess` call into the related `terminate_process_tree` and `exit_process` functions, because the functionality in each requires different permissions. To be fair, however - the EndTask binary I wrote always asked for `PROCESS_ALL_ACCESS` and I do not remember it ever being refused so long as the process being killed was created by the same user.